include/h2o/httpclient.h

```diff
@@ -173,6 +173,30 @@ typedef struct st_h2o_httpclient_timings_t {
     struct timeval response_end_at;
 } h2o_httpclient_timings_t;
 
+/**
+ * a UDP tunnel established e.g., by successful CONNECT-UDP
+ */
+typedef struct st_h2o_httpclient_udp_tunnel_t {
+    /**
+     * closes the tunnel and discards the object
+     */
+    void (*destroy)(struct st_h2o_httpclient_udp_tunnel_t *tunnel);
+    /**
+     * The write callback. Contrary to what happens in the tcp tunnel, completion is immediate.
+     */
+    void (*write_)(struct st_h2o_httpclient_udp_tunnel_t *tunnel, const void *bytes, size_t len);
```

Comment:
I think we'd want to pass list of vectors here (like `writev`). UDP packets tend to be small, and there's performance benefit by exchanging them with the kernel in batch.